query/stdlib/influxdata/influxdb/rules_test.go

```diff
@@ -122,6 +122,216 @@ func TestPushDownRangeRule(t *testing.T) {
+					plan.CreateLogicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeBy,
+						GroupKeys: []string{"_measurement", "tag0", "tag1"},
+					}),
+				},
+				Edges: [][2]int{{0, 1}},
+			},
+			After: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreatePhysicalNode("ReadGroup", &influxdb.ReadGroupPhysSpec{
+						ReadRangePhysSpec: readRange,
+						GroupMode:         flux.GroupModeBy,
+						GroupKeys:         []string{"_measurement", "tag0", "tag1"},
+					}),
+				},
+			},
+		},
+		{
+			Name: "with successor",
+			// ReadRange -> group -> count  =>  ReadGroup -> count
+			Rules: []plan.Rule{
+				influxdb.PushDownGroupRule{},
+			},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("ReadRange", &readRange),
+					plan.CreateLogicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeBy,
+						GroupKeys: []string{"_measurement", "tag0", "tag1"},
+					}),
+					plan.CreatePhysicalNode("count", &universe.CountProcedureSpec{}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{1, 2},
+				},
+			},
+			After: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreatePhysicalNode("ReadGroup", &influxdb.ReadGroupPhysSpec{
+						ReadRangePhysSpec: readRange,
+						GroupMode:         flux.GroupModeBy,
+						GroupKeys:         []string{"_measurement", "tag0", "tag1"},
+					}),
+					plan.CreatePhysicalNode("count", &universe.CountProcedureSpec{}),
+				},
+				Edges: [][2]int{{0, 1}},
+			},
+		},
+		{
+			Name: "with multiple successors",
+			//
+			// group    count       group    count
+			//     \    /       =>      \    /
+			//    ReadRange            ReadRange
+			//
+			Rules: []plan.Rule{
+				influxdb.PushDownGroupRule{},
+			},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("ReadRange", &readRange),
+					plan.CreateLogicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeBy,
+						GroupKeys: []string{"_measurement", "tag0", "tag1"},
+					}),
+					plan.CreatePhysicalNode("count", &universe.CountProcedureSpec{}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{0, 2},
+				},
+			},
+			After: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("ReadRange", &readRange),
+					plan.CreateLogicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeBy,
+						GroupKeys: []string{"_measurement", "tag0", "tag1"},
+					}),
+					plan.CreatePhysicalNode("count", &universe.CountProcedureSpec{}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{0, 2},
+				},
+			},
+		},
+		{
+			Name: "un-group",
+			// ReadRange -> group() => ReadGroup
+			Rules: []plan.Rule{
+				influxdb.PushDownGroupRule{},
+			},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("ReadRange", &readRange),
+					plan.CreateLogicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeNone,
```

Comment:
There has been confusion in the past about what `GroupModeNone` means.

The Flux definition seems to describe it as a no-op:
https://github.com/influxdata/flux/blob/master/group_mode.go#L7-L8

This would be consistent with just letting storage produce the natural grouping, leaving the output of the read grouped by tag keys.  

Is it possible to produce a `GroupModeNode` in a `GroupProcedureSpec` with Flux syntax?  I _think_ the answer is "no"---I believe that a plain `group()` in Flux will produce `GroupModeBy`, with an empty column list.

I think that we needed `GroupModeNone` in the past because there was just one read RPC, and we needed a way to express, "don't do anything with respect to grouping."  My guess is that it's obsolete now.

Reply:
Just thinking aloud, if someday `ReadGroup` can also have aggregates pushed into it, you might want `GroupModeNode` to indicate you want to aggregate over the natural grouping.

Reply:
Looking into it now. I thought GroupModeNone meant un-grouping, i.e all results in a single table. But I will confirm that.

Reply:
@wolffcm you are correct. GroupModeNone is obsolete. I will fix.